subsys/net/lib/sockets/sockets.c

```diff
@@ -209,45 +210,100 @@ int zsock_accept(int sock, struct sockaddr *addr, socklen_t *addrlen)
 	return POINTER_TO_INT(ctx);
 }
 
+static struct net_pkt* get_tx_pkt(struct net_context *ctx,
```

Comment:
this is not a getter function anymore as you are appending data as well in the process, you'll have to rename this one.